src/heap.rs

```diff
@@ -135,6 +171,7 @@ impl Index<HeapId> for Heaps {
     }
 }
 
+// Since we never remove `Heap`s (even after rollbacks), we allow additional empty heaps at the end of `Heaps`.
```

Comment:
Rollbacks do deallocate them, though, and we could store a linked list in the deallocated heaps and use that to reuse them. Maybe state more explicitly that they are not in use even though they still exist.